plonky2/src/gadgets/arithmetic.rs

```diff
@@ -206,12 +206,16 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
     }
 
     /// Multiply `n` `Target`s.
-    pub fn mul_many(&mut self, terms: &[Target]) -> Target {
-        terms
-            .iter()
-            .copied()
-            .reduce(|acc, t| self.mul(acc, t))
-            .unwrap_or_else(|| self.one())
+    pub fn mul_many<T>(&mut self, terms: impl IntoIterator<Item = T>) -> Target
+    where
+        T: Borrow<Target>,
+    {
+        let mut iter = terms.into_iter();
+        if let Some(first) = iter.next() {
```

Comment:
I think it's also fine to start with an acc of 1 since `1 * first` should be optimized away, though we can keep this style if you prefer it

Reply:
Good point, done.

Btw, what would you think of changing all the other `mul_many_extension, add_many,...` functions to also take a generic iterator instead of a slice? I think it would be more versatile and possibly save a few `collect()`s.

Reply:
Sounds good to me! I think the only downside is that the signatures in our docs become a bit harder to read, but probably worth it for ease of calling.